BarCodeScanner/ProductDetails.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useState } from 'react';
+import { View, Image, StyleSheet, Text, Button, TouchableOpacity } from 'react-native';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+type ItemData = {
+  url: string;
+  image: string | undefined;
+  title: string;
+  description: string;
+};
+
+const removeFromFavorites = async (itemUrl: string) => {
+  try {
+    const existingFavorites = await AsyncStorage.getItem('favorites');
+    const parsedFavorites = existingFavorites ? JSON.parse(existingFavorites) : [];
+
+    const updatedFavorites = parsedFavorites.filter(favorite => favorite.url !== itemUrl);
+    await AsyncStorage.setItem('favorites', JSON.stringify(updatedFavorites));
+
+    setFavorites(updatedFavorites);
+  } catch (error) {
+    console.error('Error removing from favorites:', error);
+  }
+};
+
+const addToFavorites = async (itemData: ItemData) => {
+  try {
+    const existingFavorites = await AsyncStorage.getItem('favorites');
+    const parsedFavorites = existingFavorites ? JSON.parse(existingFavorites) : [];
+
+    const isItemInFavorites = parsedFavorites.some((favorite) => favorite.image === itemData.image);
+
+    if (isItemInFavorites) {
+      alert('This item is already in your favorites.');
+      return;
+    }
+
+    
+
+    const newFavorite = {
+      title: itemData.title,
+      description: itemData.description,
+      image: itemData.image,
+      url: itemData.url,
+    };
+
+    parsedFavorites.push(newFavorite);
+    await AsyncStorage.setItem('favorites', JSON.stringify(parsedFavorites));
+
+    alert('Product added to favorites!');
+  } catch (error) {
+    console.error('Error adding to favorites:', error);
+  }
+};
+
+export const ProductDetails = ({ route }) => {
+  const { url } = route.params;
+  const [itemData, setItemData] = useState<ItemData | null>(null);
+
+  useEffect(() => {
+
+    fetch(url)
+      .then(response => response.json())
+      .then(data => {
+        setItemData(data);
+      })
+      .catch(error => {
+        console.error('Error fetching item data:', error);
+      });
+  }, []);
+
+  return (
+    <View style={styles.container}>
+      {itemData ? (
+        <>
+          <Image source={{ uri: itemData.image }} style={styles.picture} resizeMode="contain"/>
+          <Text style={styles.title}>{itemData.title}</Text>
+          <Text style={styles.desc}>{itemData.description}</Text>
+          <Button title="Add to Favorites" onPress={() => addToFavorites(itemData)} />
+          {favorites.some(favorite => favorite.url === itemData.url) && (
```

Comment:
When I try to scan a barcode, the app crashes immediately because of this line. `favorites` does not exist. Is there some missing code you forgot to push? Can you double check?